operator/controllers/reconcilers/statefulset/statefulset_reconciler.go

```diff
@@ -43,23 +45,35 @@ func NewStatefulSetReconciler(
 	podSpec *v1.PodSpec,
 	volumeClaimTemplates []mlopsv1alpha1.PersistentVolumeClaim,
 	scaling *mlopsv1alpha1.ScalingSpec,
+	serverConfigMeta metav1.ObjectMeta,
```

Comment:
Just general thought: does it matter here if we pass by pointer vs value?

Reply:
will leave for now - not sure it will make much difference but good you highlighted